test/helpers.go

```diff
@@ -0,0 +1,89 @@
+package test
+
+import (
+	"errors"
+	"io"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+// test is a home for unit test utility functions.
+
+// WithFakeFile returns a throwaway file in a test-specific directory.
+//
+// The file is automatically closed and removed when the test ends.
+func WithFakeFile(t *testing.T) (f *os.File) {
+	f, err := ioutil.TempFile(t.TempDir(), "fake-file")
+	if err != nil {
+		t.Fatalf("opening throwaway file: %s", err)
+	}
+
+	t.Cleanup(func() {
+		if err := f.Close(); err != nil {
+			t.Logf("closing throwaway file %q: %s", f.Name(), err)
+		}
+	})
+
+	return f
+}
+
+// WithFakeFileContents returns a throwaway file containing some data.
+//
+// The file is automatically closed and removed when the test ends.
+func WithFakeFileContents(t *testing.T, r io.Reader) (f *os.File) {
+	f = WithFakeFile(t)
+	_, err := io.Copy(f, r)
+	if err != nil {
+		t.Fatalf("copying contents into fake file %q: %s", f.Name(), err)
+	}
+
+	return f
+}
+
+// RemoveFileIfExists is a helper for ValidateFilename tests.
+func RemoveFileIfExists(t *testing.T, filename string) {
+	if err := os.Remove(filename); err != nil {
+		if !errors.Is(err, fs.ErrNotExist) {
+			t.Fatalf("removing file %q: %s", filename, err)
+		}
+	}
+}
+
+// UseEnvTest sets up the controller-runtime EnvTest framework.
+//
+// The test will be skipped if EnvTest framework isn't detected.
+//
+// EnvTest provides fake k8s control plane components for testing
+// purposes. The process of bringing up and tearing down the EnvTest framework
+// involves running a few binaries, and is not integrated into a vanilla "go
+// test" run, but rather can be run via the unit-test target in Makefile. See
+// https://book.kubebuilder.io/reference/envtest.html for details.
+//
+// TODO: What could be done to integrate EnvTest with go test, so that "go
+// test" would work?
+func UseEnvTest(t *testing.T) *rest.Config {
```

Comment:
Couldn't we use the `envtest` package we have in our repo?
And call it liken this
https://github.com/aws/eks-anywhere/blob/119aab6f6bceed10028b391a5d722828f8c2348d/controllers/main_test.go#L12-L14

Reply:
I think the two could be merged, but we'd probably need to have a good chat about that. Issues that come to mind immediately upon looking at newEnvironment:

1. It isn't clear if the client returned by environment can have its scheme modified, which is a requirement.
2. I have a hunch that we'd really like to be able to get our filthy paws on the *rest.Config that's returned by testenv, and that's not possible at present.
3. We don't need the manager at this time, it'd be preferable to make that optional. I already hate that I have to use testenv at all, and that it prevents me from being able to simply "go test" things. I'd prefer not to add controller startup to the things slowing down our tests.

Are you OK with it as a follow-up?

Reply:
Follow up sounds good!